x/programs/rust/wasmlanche-sdk/src/program.rs

```diff
@@ -81,11 +81,12 @@ impl<K> Program<K> {
             max_units,
         };
 
-        let args_bytes = borsh::to_vec(&args).map_err(|_| ExternalCallError::ExecutionFailure)?;
+
+        let args_bytes = borsh::to_vec(&args).expect("failed to serialize args");
 
         let bytes = unsafe { call_program(args_bytes.as_ptr(), args_bytes.len()) };
 
-        borsh::from_slice(&bytes).map_err(|_| ExternalCallError::ExecutionFailure)?
+        borsh::from_slice(&bytes).expect("failed to deserialize")
```

Comment:
I would revert this change, this way we could more easily bubble up serialization errors if needed.

Reply:
done